pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -872,20 +873,17 @@ public CompletableFuture<Void> connectionOpened(final ClientCnx cnx) {
                 log.warn("[{}][{}] Failed to subscribe to topic on {}", topic,
                         subscription, cnx.channel().remoteAddress());
 
-                if (e.getCause() instanceof PulsarClientException.TimeoutException) {
-                    // Creating the consumer has timed out. We need to ensure the broker closes the consumer
-                    // in case it was indeed created, otherwise it might prevent new create consumer operation,
-                    // since we are not necessarily closing the connection.
-                    long closeRequestId = client.newRequestId();
-                    ByteBuf cmd = Commands.newCloseConsumer(consumerId, closeRequestId);
-                    cnx.sendRequestWithId(cmd, closeRequestId);
-                }
-
                 if (e.getCause() instanceof PulsarClientException
                         && PulsarClientException.isRetriableError(e.getCause())
                         && System.currentTimeMillis() < SUBSCRIBE_DEADLINE_UPDATER.get(ConsumerImpl.this)) {
                     future.completeExceptionally(e.getCause());
                 } else if (!subscribeFuture.isDone()) {
+                    // Creating the consumer has timed out. We need to ensure the broker closes the consumer
+                    // in case it was indeed created, otherwise it might prevent new create consumer operation,
+                    // since we are not necessarily closing the connection.
+                    if (e.getCause() instanceof PulsarClientException.TimeoutException) {
+                        sendCloseConsumerCommand(cnx);
+                    }
```

Comment:
By moving this logic here, we won't _always_ send the `CLOSE_CONSUMER` protocol message after the initial connection is made, which will be problematic for timeouts that happen after load balancing. As the comment in the code suggests, the broker expects a client to send the `CLOSE_CONSUMER` command before sending a second `SUBSCRIBE` command.

I am concerned that moving this logic here could lead to unintended consequences in the client.